prac_05/emails.py

```diff
@@ -0,0 +1,31 @@
+"""
+Emails
+Estimate: 15 minutes
+Actual: 20 minutes
+"""
+
+
+def main():
+    """Build dictionary of emails and names"""
+    email_to_name = {}
+    email = input("Email: ")
+    while email != '':
+        name = extract_name_from_email(email)
+        name_check = input(f"Is your name {name}? (Y/n) ").upper()
+        if name_check != 'Y' and name_check != '':
+            name = input("Name: ").title()
+        email_to_name[email] = name
+        email = input("Email: ")
+    for email, name in email_to_name.items():
+        print(f"{email} {name}")
+
+
+def extract_name_from_email(email):
+    """Extract name from before @ symbol in email"""
+    email_prefix = email.split('@')[0]
```

Comment:
I don't think we have covered using [0] in that case yet, be careful using it. Granted, it is much more efficient...

Reply:
Additionally, you can combine the below line with this one if you do something like: 

split_name = (user_email[:user_email.find('@')]).split(".")

This reads up until the line index of the '@' and then splits the result at "."